src/org/joni/Matcher.java

```diff
@@ -325,7 +339,13 @@ private final int searchCommon(int start, int range, int option, boolean interru
                 /* search start-position only */
                 // !begin_position:!
                 if (range > start) {
-                    range = start + 1;
+                    if (gpos > start) {
```

Comment:
I think https://github.com/jruby/joni/blob/master/src/org/joni/Matcher.java#L370 will also need that logic